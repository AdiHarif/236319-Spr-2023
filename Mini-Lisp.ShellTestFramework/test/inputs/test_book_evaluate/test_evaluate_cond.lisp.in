; =================================================
; evaluate-cond - 1 arg
; =================================================
(evaluate-cond '() '(t.t nil.nil))              ; NIL
(evaluate-cond '((t nil)) '(t.t nil.nil))       ; NIL
(evaluate-cond '((a t)) '(a.t t.t nil.nil))     ; T
(evaluate-cond '((a b)) '(a.t b.t t.t nil.nil)) ; T

; =================================================
; evaluate-cond - 2 arg
; =================================================
(evaluate-cond '((t a) (nil b)) '(a.a b.b t.t nil.nil)) ; A
(evaluate-cond '((nil a) (t b)) '(a.a b.b t.t nil.nil)) ; B
(evaluate-cond '((a b) (b a)) '(a.t b.t t.t nil.nil))   ; T
(evaluate-cond '((a b) (b a)) '(a.nil b.t t.t nil.nil)) ; NIL

; =================================================
; evaluate-cond - more args
; =================================================
(evaluate-cond '((a b) (a b) (b a)) '(a.nil b.t t.t nil.nil)) ; NIL

; =================================================
; evaluate-cond - errors
; =================================================
(evaluate-cond '(t t) '(t.t nil.nil))   ; argument car error
(evaluate-cond '((t a)) '(t.t nil.nil)) ; undefined
