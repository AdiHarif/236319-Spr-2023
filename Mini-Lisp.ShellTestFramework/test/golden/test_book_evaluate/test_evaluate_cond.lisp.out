NIL
NIL
T
T
A
B
T
NIL
NIL
Traceback (most recent call last):
	From function EVALUATE-COND:
		Args: ((T T) ((T.T) (NIL)))
	From function COND:
		Args: (((NULL TEST-FORMS) NIL) ((EVALUATE (CAR (CAR TEST-FORMS)) A-LIST) (EVALUATE (CAR (CDR (CAR TEST-FORMS))) A-LIST)) (T (EVALUATE-COND (CDR TEST-FORMS) A-LIST)))
	From function EVALUATE:
		Args: ((CAR (CAR TEST-FORMS)) A-LIST)
	From function CAR:
		Args: T
Error [CAR] in T
Traceback (most recent call last):
	From function EVALUATE-COND:
		Args: (((T A)) ((T.T) (NIL)))
	From function COND:
		Args: (((NULL TEST-FORMS) NIL) ((EVALUATE (CAR (CAR TEST-FORMS)) A-LIST) (EVALUATE (CAR (CDR (CAR TEST-FORMS))) A-LIST)) (T (EVALUATE-COND (CDR TEST-FORMS) A-LIST)))
	From function EVALUATE:
		Args: (A ((T.T) (NIL)))
	From function COND:
		Args: (((ATOM S-EXPRESSION) (LOOKUP S-EXPRESSION A-LIST)) ((IS-ATOMIC (CAR S-EXPRESSION)) (EVALUATE-ATOMIC S-EXPRESSION A-LIST)) (T (APPLY (EVALUATE (CAR S-EXPRESSION) A-LIST) (CDR S-EXPRESSION) A-LIST)))
	From function LOOKUP:
		Args: (A ((T.T) (NIL)))
	From function COND:
		Args: (((NULL A-LIST) (ERROR UNDEFINED ID)) ((EQ ID (CAR (CAR A-LIST))) (CDR (CAR A-LIST))) (T (LOOKUP ID (CDR A-LIST))))
	From function LOOKUP:
		Args: (A ((NIL)))
	From function COND:
		Args: (((NULL A-LIST) (ERROR UNDEFINED ID)) ((EQ ID (CAR (CAR A-LIST))) (CDR (CAR A-LIST))) (T (LOOKUP ID (CDR A-LIST))))
	From function LOOKUP:
		Args: (A NIL)
	From function COND:
		Args: (((NULL A-LIST) (ERROR UNDEFINED ID)) ((EQ ID (CAR (CAR A-LIST))) (CDR (CAR A-LIST))) (T (LOOKUP ID (CDR A-LIST))))
	From function ERROR:
		Args: UNDEFINED
Error [UNDEFINED] in UNDEFINED
